sdk/src/wallet/utxo.ts

```diff
@@ -0,0 +1,159 @@
+import { Transaction, Script, selectUTXO, TEST_NETWORK, NETWORK, p2wpkh, p2sh } from '@scure/btc-signer';
+import { hex } from "@scure/base";
+import { AddressType } from 'bitcoin-address-validation';
+
+import { DefaultEsploraClient, UTXO } from '../esplora';
+
+export type BitcoinNetworkName = 'mainnet' | 'testnet';
+
+const bitcoinNetworks: Record<BitcoinNetworkName, typeof NETWORK> = {
+  mainnet: NETWORK,
+  testnet: TEST_NETWORK
+};
+
+export const getBtcNetwork = (name: BitcoinNetworkName) => {
+  return bitcoinNetworks[name];
+};
+
+type Output = { address: string; amount: bigint } | { script: Uint8Array; amount: bigint };
+
+export interface Input {
+  txid: string;
+  index: number;
+  witness_script?: Uint8Array;
+  redeemScript?: Uint8Array;
+  witnessUtxo?: {
+    script: Uint8Array;
+    amount: bigint;
+  };
+  nonWitnessUtxo?: Uint8Array;
+}
+
+export async function createTransfer(
+  network: BitcoinNetworkName,
+  addressType: AddressType,
+  fromAddress: string,
+  toAddress: string,
+  amount: number,
+  publicKey?: string,
+  opReturnData?: string,
+  confirmationTarget: number = 3,
+): Promise<Transaction> {
+  const esploraClient = new DefaultEsploraClient(network);
+
+  // NOTE: esplora only returns the 25 most recent UTXOs
+  // TODO: change this to use the pagination API and return all UTXOs
```

Comment:
Do you want to address this before merging? Or maybe make an issue for it?

Reply:
Will make an issue